docs/separate-benchmark-module.md

```diff
@@ -0,0 +1,102 @@
+# Creating Separate Module for Benchmarks
+
+This guide will walk you through the process of creating a separate module for benchmarks in your Kotlin project. 
+Creating a separate module for benchmarks is especially beneficial when you are integrating benchmarks into an existing project. 
+Here are a few advantages of doing so:
+
+1. **Organization**: It helps maintain a clean and organized project structure. Segregating benchmarks from the main code makes it easier to navigate and locate specific code segments.
+
+2. **Isolation**: Separating benchmarks ensures that the benchmarking code does not interfere with your main code or test code. This isolation guarantees accurate measurements without unintentional side effects.
+
+3. **Flexibility**: Creating a separate module allows you to manage your benchmarking code independently. You can compile, test, and run benchmarks without impacting your main source code.
+
+## Step-by-step Setup Guide
+
+Below are the step-by-step instructions to set up a separate module for benchmarks in both Kotlin Multiplatform and Kotlin/JVM projects:
+
+### Kotlin Multiplatform Project
+
+Follow these steps to set up a separate compilation for benchmarks:
+
+1. **Define New Compilation**:
+
+    Start by creating a new compilation in your target of choice (e.g. jvm, js, native, wasm etc.). 
```

Comment:
Is it something a user always have to do when creating a separate module with benchmarks or it's some advanced feature? If it's the latter then when should a user configure it?
I've added several separate benchmarking modules (that are using `kotlinx-benchmarks`) into several separate projects, but I have never created separate compilations. It was always "add a module, apply plugin, add dependency on a module under test, register targets".

Reply:
As we figured it out, the term "module" is about compilations, so one has to set it up. ;)

Reply:
Yes, it is considered an advanced feature. It is needed when one wants to use/access/benchmark internal API of an existing project.
As I described [above](https://github.com/Kotlin/kotlinx-benchmark/pull/175#discussion_r1436951614), the basic scenario is to create a separate project (or subproject) that is dedicated to benchmarks.